testsuite/pytests/test_parameter_operators.py

```diff
@@ -94,37 +96,32 @@ def test_binary_operators(op, a, b):
     Outer parametrization is over operators, the inner over param-param, param-number and number-param combinations.
     """
 
-    # Extract underlying numerical value if we are passed a parameter object.
-    try:
-        val_a = a.GetValue()
-    except AttributeError:
-        val_a = a
-    try:
-        val_b = b.GetValue()
-    except AttributeError:
-        val_b = b
+    val_a = to_numeric(a)
+    val_b = to_numeric(b)
 
     assert op(a, b).GetValue() == op(val_a, val_b)
 
 
-@pytest.mark.parametrize('op, a, b', [
-    [ops.pow, const_param(31), 5]])
-def test_incomplete_binary_operators(op, a, b):
+@pytest.mark.parametrize('op, a, b, is_supported', [
+    [ops.pow, _const_param(31), 5, True],
+    [ops.pow, _const_param(31), _const_param(5), False],
+    [ops.pow, 31, _const_param(5), False],
+    ])
+def test_incomplete_binary_operators(op, a, b, is_supported):
     """
-    Perform tests for binary operators that do not support Parameter as any operand.
+    Perform tests for binary operators that only support Parameters as one operand.
+
+    Pass `False` for `is_supported` to indicate unsupported operand combinations.
```

Comment:
Can you please elaborate on "indicate" here? From the current description it does not really become clear what the function would do if `True` or `False` is passed. Something like:

> If `True` is passed, the function asserts that the operator works as expected, if `False` is passed, it checks if an Exception is raised.